agent/engine/host_resource_manager.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package engine contains the core logic for managing tasks
+
+package engine
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+)
+
+// HostResourceManager keeps account of each task in
+type HostResourceManager struct {
+	hostResource     map[string]*ecs.Resource
+	consumedResource map[string]*ecs.Resource
+
+	taskConsumed map[string]bool //task.arn to boolean whether host resources consumed or not
+}
+
+func NewHostResourceManager(resourceMap map[string]*ecs.Resource) HostResourceManager {
+	consumedResourceMap := make(map[string]*ecs.Resource)
+	taskConsumed := make(map[string]bool)
+	// assigns CPU, MEMORY, PORTS, PORTS_UDP from host
```

Comment:
Nit - is it possible to make `PORTS` -> `PORTS_TCP` just to be clear?

Reply:
Sure, will change it